application/models/services/Quotas.php

```diff
@@ -546,22 +547,27 @@ static public function checkCompletedQuota(int $surveyid, array $updatedValues =
         $sMessage = $event->get('message', $aMatchedQuota['quotals_message']);
         $sUrl = $event->get('url', $aMatchedQuota['quotals_url']);
         $sUrlDescription = $event->get('urldescrip', $aMatchedQuota['quotals_urldescrip']);
-        $sAction = $event->get('action', $aMatchedQuota['action']);
+        $sAction = (int) $event->get('action', $aMatchedQuota['action']);
         /* Tag if we close or not the survey */
-        $closeSurvey = ($sAction == "1" || App()->getRequest()->getPost('move') == 'confirmquota');
+        $closeSurvey = ($sAction === 1 || App()->getRequest()->getPost('move') === 'confirmquota');
```

Comment:
We have the same issue here, but it's really for `$sAction === 2` only, all others close survey.

Then : maybe : `$closeSurvey = ($sAction !== 2 || App()->getRequest()->getPost('move') === 'confirmquota');` ?

Reply:
yep, fixed, wasnt using static var